var/spack/repos/builtin/packages/netcdf/package.py

```diff
@@ -160,6 +162,9 @@ def configure_args(self):
             # possible to use this library in shared builds
             cflags.append(self.compiler.pic_flag)
 
+        if '+pic' in self.spec:
```

Comment:
@bebosudo could you add a comment here like

`netcdf always builds static libraries, this ensures that the static library is built with pic, including when shared libraries are also built`

This could be combined with the above `if` like:

```
if '+pic' in self.spec or '~shared' in self.spec:
  cflags.append(self.compiler.pic_flag)
```

> I didn't touch anything inside the netcdf package, because there are lots of components.

That's fine